Source/Events.Store.MongoDB/Processing/Streams/StreamProcessorState.cs

```diff
@@ -17,21 +18,40 @@ public class StreamProcessorState
         /// <summary>
         /// Initializes a new instance of the <see cref="StreamProcessorState"/> class.
         /// </summary>
-        /// <param name="id">The <see cref="StreamProcessorId" />.</param>
+        /// <param name="scopeId">The <see cref="ScopeId" />.</param>
+        /// <param name="eventProcessorId">The <see cref="EventProcessorId" />.</param>
+        /// <param name="sourceStreamId">The <see cref="SourceStreamId" />.</param>
         /// <param name="position">The position.</param>
         /// <param name="failingPartitions">The states of the failing partitions.</param>
-        public StreamProcessorState(StreamProcessorId id, ulong position, IDictionary<string, FailingPartitionState> failingPartitions)
+        public StreamProcessorState(Guid scopeId, Guid eventProcessorId, Guid sourceStreamId, ulong position, IDictionary<string, FailingPartitionState> failingPartitions)
         {
-            Id = id;
+            ScopeId = scopeId;
+            EventProcessorId = eventProcessorId;
+            SourceStreamId = sourceStreamId;
             Position = position;
             FailingPartitions = failingPartitions;
         }
 
         /// <summary>
-        /// Gets or sets the <see cref="StreamProcessorId" />.
+        /// Gets or sets the  MongoDB _id. This is used so that the class would have a valid '_id' field in mongo.
+        /// The classes 'true' id is comporomised from the combinaton of ScopeId, EventProcessorId and SourceStreamId.
+        /// </summary>
+        public ObjectId Id { get; set; }
```

Comment:
I think that this is OK for now, but as you said there may be another solution. I'm not familiar enough with MongoDB to know